ElementX/Sources/Screens/Other/GenericCallLinkCoordinator.swift

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright 2023 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+import WebKit
+
+struct GenericLinkCoordinatorParameters {
+    let url: URL
+}
+
+class GenericCallLinkCoordinator: CoordinatorProtocol {
+    private let parameters: GenericLinkCoordinatorParameters
+    
+    init(parameters: GenericLinkCoordinatorParameters) {
+        self.parameters = parameters
+    }
+    
+    func toPresentable() -> AnyView {
+        AnyView(
+            WebView(url: parameters.url)
+                .ignoresSafeArea(edges: .bottom)
+                .presentationDragIndicator(.visible)
+        )
+    }
+}
+
+private struct WebView: UIViewRepresentable {
+    let url: URL
+
+    func makeUIView(context: Context) -> WKWebView {
+        context.coordinator.webView
+    }
+
+    func makeCoordinator() -> Coordinator {
+        Coordinator()
+    }
+
+    func updateUIView(_ webView: WKWebView, context: Context) {
+        webView.load(URLRequest(url: url))
+    }
+
+    @MainActor
+    class Coordinator: NSObject, WKUIDelegate {
+        private(set) var webView: WKWebView!
+
+        override init() {
+            super.init()
+            
+            let configuration = WKWebViewConfiguration()
+
+            configuration.allowsInlineMediaPlayback = true
+            configuration.allowsPictureInPictureMediaPlayback = true
+            
+            webView = WKWebView(frame: .zero, configuration: configuration)
+            webView.uiDelegate = self
+        }
+
+        // MARK: - WKUIDelegate
+        
+        func webView(_ webView: WKWebView, decideMediaCapturePermissionsFor origin: WKSecurityOrigin, initiatedBy frame: WKFrameInfo, type: WKMediaCaptureType) async -> WKPermissionDecision {
+            .grant
```

Comment:
*please* can we check the origin before blindly streaming the users camera to the page presented here.
```suggestion
            guard origin.host == appSettings.elementCallURL.host else { return .deny }
            .grant
```

Reply:
Could also add the navigation delegate to make sure you don't end up with a link on the page taking you outside the call:

https://github.com/pixlwave/Element-Call-macOS/blob/6407a23458ae95b402922e4ebfdb357b4ecef819/Sources/VOIPCoordinator.swift#L118-L131

Reply:
Good idea 👍

Reply:
Make sense 👍